front/src/applications/stdcm/utils/addSecondaryCodesToRefSchedules.ts

```diff
@@ -0,0 +1,30 @@
+import type { StdcmPathStep } from 'reducers/osrdconf/types';
+
+import type { RefSchedule, RefScheduleWithSecondaryCode } from '../types';
+
+export const addSecondaryCodesToRefSchedules = (
+  refSchedules: RefSchedule[],
+  pathSteps: StdcmPathStep[]
+): RefScheduleWithSecondaryCode[] => {
+  const nameToSecondaryCode = new Map<string, string>();
+
+  pathSteps.forEach((step) => {
+    const loc = step.location;
+    if (loc?.name && loc?.secondary_code) {
+      nameToSecondaryCode.set(loc.name, loc.secondary_code);
+    }
+  });
+
+  return refSchedules.map((schedule) => ({
+    train_id: schedule.train_id,
+    start_time: schedule.start_time,
+    begin: {
+      ...schedule.begin,
+      secondary_code: nameToSecondaryCode.get(schedule.begin.id) ?? '—',
```

Comment:
Are you manipulating ids or names in this function (nameToSecondaryCode but schedule.begin.id) ?

If the issue is in the backend, can you discuss it with backend developers to fix this ? (it can be done after your PR of course, but we should leave a TODO in this case)